httpclient/response.go

```diff
@@ -25,6 +25,17 @@ func WithResponseHeader(key string, value *string) DoOption {
 	}
 }
 
+// WithResponseUnmarshal unmarshals the response body into response. The
+// supported response types are the following:
+//   - *bytes.Buffer,
+//   - *io.ReadCloser,
+//   - *[]byte,
+//   - a pointer to a struct with a Contents io.ReadCloser field,
+//   - a pointer to a struct representing a JSON object.
+//
+// If response is a pointer to a io.ReadCloser or a struct with a io.ReadCloser
+// field name "Contents", then the response io.ReadCloser is set to the value of
```

Comment:
The key thing to note here is that currently, our autogeneration _only_ includes a `Contents io.ReadCloser` field for APIs that return non-JSON responses. We don't attempt to parse these at all and simply return them to the caller.